core/src/main/scala/org/broadinstitute/dsde/rawls/metrics/RawlsExpansion.scala

```diff
@@ -35,33 +34,47 @@ object RawlsExpansion {
 
   /**
     * Creates an expansion for Uri which redacts pieces of the Uri matched by the provided PathMatcher.
-    * @param pathMatcher a PathMatcher which matches 1 or more String elements. If the PathMatcher matches,
+    * @param pathMatchers seq of PathMatcher which matches 1 or more String elements. If the PathMatcher matches,
     *                    any matched Strings will be redacted from the Uri. Otherwise, the Uri is unchanged.
     * @return Uri Expansion instance
     */
-  def redactedUriExpansion[L <: HList](pathMatcher: PathMatcher[L]): Expansion[Uri] = {
+  def redactedUriExpansion[L <: Product](pathMatchers: Seq[PathMatcher[L]]): Expansion[Uri] = {
 
     new Expansion[Uri] {
       override def makeName(uri: Uri): String = {
-        // Apply the provided PathMatcher. This will give us a HList of extractions, if it matches the Path.
```

Comment:
had to rewrite this because spay used shapeless but akka http dropped that library. The new code has to use tuples instead of HLists. Variable length tuples are difficult to work with generically but they all extend Product so we use that.